dht.go

```diff
@@ -73,6 +73,9 @@ type IpfsDHT struct {
 	alpha      int // The concurrency parameter per path
 	d          int // Number of Disjoint Paths to query
 
+	queryPeerFilter        func(h host.Host, ai peer.AddrInfo) bool
+	routingTablePeerFilter func(conns []network.Conn) bool
```

Comment:
Should these have named function types, if so should they be exported or unexported?

Reply:
given the multiple implementations, it seems like not a bad idea to define types for these.

Reply:
@Stebalien I'd like to pass the DHT too. While I can pass the DHT here, it's largely irrelevant because I can't pass the DHT into an option due to circular dependencies (DHT pkg depends on options package, so options package can't use the DHT struct).

Reply:
Lol... Ok, nevermind.

(maybe we should move the options back into the DHT package but we can deal with that breaking change later).

Reply:
is there anything other than `host` that we want to have access to from passing the full DHT into the filters / option?
we can make an interface to break the circular dependency if it's useful.